### PR TITLE
#58: generator to select randomly from a fixed set - solved but in integ...

### PR DESCRIPTION
**Pull request by jgheerardyn**
...er random number generator - tests

### CODE REVIEW

File: foundations/src/main/java/io/yields/math/framework/data/DiscreteRandomGenerator.java [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+ * generates samples randomly chosen from a set.
+ */
+public class DiscreteRandomGenerator<T> implements RandomGenerator<T> {
+
+    private RandomGenerator<Integer> indexGenerator;
+    private List<T> samples;
+
+    public DiscreteRandomGenerator(List<T> samples, int seed) {
+        isTrue(samples != null, "List of samples should not be null.");
+        isTrue(!samples.isEmpty(), "List of samples shoule not be empty.");
```
Comment (rombautt) [inline foundations/src/main/java/io/yields/math/framework/data/DiscreteRandomGenerator.java]: typo, shoule

File: foundations/src/main/java/io/yields/math/framework/range/IntegerRange.java [hunk 2]
```diff
@@ -25,6 +27,7 @@
 
     public IntegerRange(Integer min, Integer max, boolean minOpen, boolean maxOpen) {
         super(min, max, minOpen, maxOpen);
+        Validate.isTrue(!(min == max - 1 && minOpen && maxOpen), "Interval cannot be empty.");
```
Comment (rombautt) [inline foundations/src/main/java/io/yields/math/framework/range/IntegerRange.java]: a little hard to read, can't this be split in 3 validates?

File: foundations/src/test/java/io/yields/math/framework/data/DiscreteRandomGeneratorTest.java [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+import java.util.stream.IntStream;
+
+import static org.fest.assertions.Assertions.assertThat;
+
+;
```
Comment (rombautt) [inline foundations/src/test/java/io/yields/math/framework/data/DiscreteRandomGeneratorTest.java]: remove

File: foundations/src/test/java/io/yields/math/framework/data/DiscreteRandomGeneratorTest.java [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+
+public class DiscreteRandomGeneratorTest {
+
+    @Test(expected = IllegalArgumentException.class)
+    public void constructor_nullList() {
```
Comment (rombautt) [inline foundations/src/test/java/io/yields/math/framework/data/DiscreteRandomGeneratorTest.java]: rename to constructor_samplesAreNull

File: foundations/src/test/java/io/yields/math/framework/data/DiscreteRandomGeneratorTest.java [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+        new DiscreteRandomGenerator<>(null, 0);
+    }
+
+    @Test(expected = IllegalArgumentException.class)
+    public void constructor_emptyList() {
```
Comment (rombautt) [inline foundations/src/test/java/io/yields/math/framework/data/DiscreteRandomGeneratorTest.java]: same as above

File: foundations/src/main/java/io/yields/math/framework/range/IntegerRange.java (hunk 2 above)
Comment (jgheerardyn) [inline foundations/src/main/java/io/yields/math/framework/range/IntegerRange.java]: not really, the point here is that the interval is empty only when the three conditions are met. Will add docs

File: foundations/src/test/java/io/yields/math/framework/data/RandomIntGeneratorTest.java [hunk 6]
```diff
@@ -0,0 +1,56 @@
...
+
+import static org.fest.assertions.Assertions.assertThat;
+
+public class RandomIntGeneratorTest {
+
+    @Test
+    public void testRange() {
+        RandomGenerator<Integer> generator = new RandomIntGenerator().withRange(new IntegerRange(0, 10, false, true));
+        Collection<Integer> allInts = new HashSet<>();
+        IntStream.range(0, 100).forEach(index -> {
```
Comment (rombautt) [inline foundations/src/test/java/io/yields/math/framework/data/RandomIntGeneratorTest.java]: better use:

IntStream.range.map(index -> {

int sample = 
assertThat()
return sample;
}).collect(toSet)

**Comment by rombautt** [comment]
Reviewed, feel free to merge.

File: foundations/src/main/java/io/yields/math/framework/range/IntegerRange.java (hunk 2 above)
Comment (rombautt) [inline foundations/src/main/java/io/yields/math/framework/range/IntegerRange.java]: k, and what about Validate.isFalse, iso isTrue with a negated condition

**Pull request merged**